### ISSUE TITLE
Fix flaky test MultipleStargateInstancesTest.shouldDistributeTrafficUniformly

### ISSUE THREAD

**Issue by jorgebay** [stargate/stargate#232]
The test `shouldDistributeTrafficUniformly` is failing often on CI.

Example:

```
Error:  Failures: 
Error:    MultipleStargateInstancesTest.shouldDistributeTrafficUniformly:117 
Expected size:<3> but was:<1> in:
<[Node(endPoint=/127.0.0.13:9043, hostId=a2e98542-13bd-4108-97b9-436fa05bf358, hashCode=7cba991a)]>
[INFO] 
Error:  Tests run: 195, Failures: 1, Errors: 0, Skipped: 3
```

**Comment by olim7t** [comment]
It fails because the driver has not discovered all 3 Stargate nodes:
```java
Collection<Node> nodes = session.getMetadata().getNodes().values();
assertThat(nodes).hasSize(runningStargateNodes);
```
That information is initialized from `system.peers`. It's not a connection issue, all nodes should be there, even if the driver has not successfully connected to all of them.

**Comment by olim7t** [comment]
It looks like system.peers gets updated, but the driver doesn't see the changes. From the logs I added in the linked PR:
```
BaseOsgiIntegrationTest.java:156 - Expected: 3, in driver metadata: 1, in system tables: 3
```

I think I've found the root cause of the problem. Here's the topology event received by the driver:
```
ControlConnection.java:179 - [s0] Processing incoming event EVENT TOPOLOGY_CHANGE(NEW_NODE /127.0.0.11:7000)
```
The port is wrong. The driver expects the RPC port (e.g. 9042), not the inter-node communication port.

**Comment by olim7t** [comment]
Opened #250 for the event bug.

**Comment by tomekl007** [comment]
> It looks like system.peers gets updated, but the driver doesn't see the changes. From the logs I added in the linked PR:
> 
> ```
> BaseOsgiIntegrationTest.java:156 - Expected: 3, in driver metadata: 1, in system tables: 3
> ```
> 
> I think I've found the root cause of the problem. Here's the topology event received by the driver:
> 
> ```
> ControlConnection.java:179 - [s0] Processing incoming event EVENT TOPOLOGY_CHANGE(NEW_NODE /127.0.0.11:7000)
> ```
> 
> The port is wrong. The driver expects the RPC port (e.g. 9042), not the inter-node communication port.

Could you elaborate more on that? If the port is incorrect, why this test is getting all N - 1 node from `system.peers` but sometimes it fails? If this is an incorrect config, should this test always fail?

**Comment by olim7t** [comment]
It's a timing issue: we start the Stargate nodes, then create the driver session with the first node as the contact point.
* If the contact point's `system.peers` has been updated before the driver starts, the information it contains is correct and everything goes well.
* But if the information has not replicated yet, `system.peers` is still empty and the driver sees only one node. The peers get added later, and the driver should catch up thanks to topology events. My goal with this PR was to wait for those events. But the driver looks for a row that is an exact match for the event's address. Since the port is wrong, it finds nothing and the event is ignored.

I was able to reproduce this systematically with a class that starts one Stargate node, creates a driver session, and then starts a second Stargate node. The driver never sees the second node. However if I create another session at that point, it sees both nodes.

To fix this test we need both #250 to get the right data in events, and this PR to wait for those events.
